### PR TITLE
fix: Reformat  gpu_is_available() function

### LINKED ISSUE #23132: gpu_is_available

### PR DESCRIPTION
**Pull request by Boghdady9**
The function `gpu_is_available()` now correctly determines whether a GPU is available.

# PR Description

## Related Issue

Close #23132

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you follow the steps we provided?

### CODE REVIEW

**Comment by Ookamice** [comment]
@Boghdady9 
`.gpu_is_avaliable()` is correct.

I'll be closing this PR and I hope you go through the procedure of opening an issue before opening a PR next time.😃

**Comment by Boghdady9** [comment]
is the function reformated correctly ?. if it's correct, why the issue is not marked as done  in the ToDo list @Ookamice

**Comment by Boghdady9** [comment]
this is done in Development section right?

**Comment by Ookamice** [comment]
@Boghdady9 
Hiya, im not sure what ToDo list you're referring to, could you link an issue number so I can have a look?

**Comment by Boghdady9** [comment]
@Ookamice 
I've just linked it

File: ivy/functional/ivy/device.py [hunk 1]
```diff
@@ -689,15 +689,16 @@ def gpu_is_available() -> bool:
...
 
     Examples
     --------
-    >>> print(ivy.gpu_is_available())
+    >>> print(gpu_is_available())
     False
     """
-    return ivy.current_backend().gpu_is_available()
+
+    return ivy.current_backend().is_gpu_available()
```
Comment (Ookamice) [inline ivy/functional/ivy/device.py:701]: This is not correct, the backend bindings for this function should remain is `gpu_is_available()`

File: ivy/functional/ivy/device.py [hunk 2]
```diff
@@ -689,15 +689,16 @@ def gpu_is_available() -> bool:
...
     -------
-    ret
-        Boolean, as to whether a gpu is available.
+    ret: bool
+        Boolean, indicating whether a GPU is available.
 
     Examples
     --------
-    >>> print(ivy.gpu_is_available())
+    >>> print(gpu_is_available())
```
Comment (Ookamice) [inline ivy/functional/ivy/device.py:697]: The ivy prefix is still needed in the examples

File: ivy/functional/ivy/device.py [hunk 3]
```diff
@@ -689,15 +689,16 @@ def gpu_is_available() -> bool:
 
     Returns
     -------
-    ret
-        Boolean, as to whether a gpu is available.
+    ret: bool
```
Comment (Ookamice) [inline ivy/functional/ivy/device.py:692]: The type hint here is unnecessary here and to be consistent with other typehints in this document, please revert this change.

File: ivy/functional/ivy/device.py [hunk 4]
```diff
@@ -689,15 +689,16 @@ def gpu_is_available() -> bool:
...
+        Boolean, indicating whether a GPU is available.
```
Comment (Ookamice) [inline ivy/functional/ivy/device.py:693]: This is just a matter of English semantics, I believe yours flow better so feel free to keep this.

**Review by Ookamice — COMMENTED** [review]
@Boghdady9
Just read through your linked issue. From my understanding this should be a task about updating the backend implementations of `gpu_is_available()`  as well as updating the doctring to be more informative.

Since the function itself is fine -I personally haven't dug through the backend implementation but I'll take Anna's word for it in another PR you have opened regarding the same issue-, I believe the only change necessary here is line 693. Please revert all the other changes you have made and I'll merge this. Thanks.

**Comment by Boghdady9** [comment]
@Ookamice 
I Made all what you said in the second commit, so please merge it

**Comment by Ookamice** [comment]
@Boghdady9 
Hiya, im seeing 0 files changed in the github tab? Perhaps you've rolled everything back?
Also I dug through the git history a little and found that https://github.com/unifyai/ivy/issues/1230 was already informed that this issue has been closed.

**Comment by Boghdady9** [comment]
@Ookamice 

but when I chosen it, it wasn't unmarked and seemed that no one took it before😢.. I already committed the changes, when I go to commit I found two commits Commits on Sep 7, 2023 and the other  on Sep 8, 2023 
Screenshot 2023-09-08 at 16 47 05

**Comment by Ookamice** [comment]
@Boghdady9 
You've modified 4 lines on Sep 7, 2023 and then you've modified the same 4 lines back to its original state on Sep 8, 2023. Hence the total change (in the Files changed tab) shows 0.
